### ISSUE TITLE
Simplify (or remove) handling of multiple winit versions

### ISSUE THREAD

**Issue by dbr** [imgui-rs/imgui-rs#627]
I find the imgui-winit-support module extremely hard to follow, as there are methods duplicated several times, each with differing `#[cfg(...)]`

I propose we basically have one file per version of winit. E.g we have `winit_25.rs`, `winit_26.rs` etc - and `lib.rs` just brings in the right one based on the same `#[cfg(...)]` checks

The reasoning being:

1. It's makes it easier to diff two files to find what changes between each version (e.g `diff winit_25.rs winit_26.rs`)
2. While it increases the amount of duplication, it's not a lot - a majority of the methods are already duplicated (basically just `WinitPlatform::init`, `WinitPlatform::copy_mouse_to_io` and `WinitPlatform::prepare_render` are not duplicated)
3. It contains the `#[cfg(...)]` to a smaller area, which might make adding new versions simpler and less accident prone, same with dropping support for older versions

@thomcc @sanbox-irl - thoughts?

**Comment by dbr** [comment]
Releatedly: what versions of winit do we still practically need to handle?

I realized I don't actually understand why we have support for so many winit versions - poking around it seems to originate from #228 which added support for 0.20 which was a big breaking change from 0.19.. but the description accompanying it was interesting:

> The main reason for supporting both versions is that gfx-rs is still on 0.19, so we can't just upgrade to 0.20. I've got a hunch we'll need to drop gfx-rs support soon completely because it's not maintained actively...

..and we have long since dropped gfx-rs as the project is abandonded

I wonder if a more tenable policy is:

1. Support the current primary version of winit at time of release
2. Also support the previous primary version of winit from last

So for imgui-rs 0.9 we suport winit-0.25 (default in imgui-rs 0.8) plus winit-0.26 (current release). For 0.10 we would support only 0.26 (assuming no major winit release before then)

That way we provide a smoother upgrade path (user can update imgui, *then* winit - rather than requiring both at once), while keeping the maintenance burdent reasonable (i.e not supporting 2019 release of winit)

However I wonder how necessary even this is - I'm not aware of (m)any crates which support multiple versions of their dependencies as a general policy

**Comment by dbr** [comment]
I made #638 to see how things look if we just delete the old-winit-feature-flags

Given:

1. the fairly minor changes in `imgui-winit-support` (I think there was code differences only for winit-0.19 and 0.20, all subsequent versions were about the same)
2. the initial cause for this old-version support was the subsequently abandonded gfx-rs crate
3. that `imgui-winit-support` is most probably a "leaf dependency"

..I'm increasingly tempted to go the "just support the current release" approach, and if someone requires both an old version of winit *and* the latest imgui-rs, I feel it is not overly burdonsome for them to maintain a temporary fork of `imgui-winit-support`

That said, I might be overlooking some major crate which is stuck on winit-0.25?

**Comment by thomcc** [comment]
According to https://crates.io/crates/winit most things are on latest version. For a long time things were stuck on 0.18 I think. I don't do much in this crate anymore, but FWIW I'm in favor at this point of removing all the old versions, I don't think it's worth keeping anymore (and I've also learned more about how Cargo version resolution works, and no longer think this is a good approach).

**Comment by sanbox-irl** [comment]
I also agree! we had a problem because winit got split in two after ~0.20, but most are current. I say, sorry, you don't get current imgui-rs support!

**Comment by dbr** [comment]
Merged :partying_face:
